client/containers/Pub/SpubHeader/PreviewTab/SubmitDialog.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from 'react';
+import { Fragment } from 'prosemirror-model';
+
+import { Callout, Button, Classes, Dialog } from '@blueprintjs/core';
+import { apiFetch } from 'client/utils/apiFetch';
+import { Submission, SubmissionStatus, DefinitelyHas } from 'types';
+import { Editor } from 'components';
+import { getEmptyDoc, jsonToNode } from 'components/Editor';
+import { usePubContext } from 'containers/Pub/pubHooks';
+
+type Props = {
+	submission: DefinitelyHas<Submission, 'submissionWorkflow'>;
+	isOpen: boolean;
+	onClose: () => any;
+};
+
+const SubmitDialog = (props: Props) => {
+	const { collabData } = usePubContext();
+	const { view } = collabData.editorChangeObject;
+	const [isHandlingSubmission, setIsHandlingSubmission] = useState(false);
+	const [updatedSubmission, setUpdatedSubmission] = useState(null);
+	const [submissionErr, setSubmissionErr] = useState(null);
+	const onSubmit = () => {
+		const abstract = props.submission.abstract || getEmptyDoc();
```

Comment:
For cleanliness consider breaking the Prosemirror manipulation out into its own local helper function:

```ts
addAbstractToEditor(abstract: DocJson, view: EditorView): void
```